src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/ManagedRuleOverride/AzureApplicationGatewayFirewallPolicyManagedRuleOverride.cs

```diff
@@ -43,6 +43,11 @@ public class AzureApplicationGatewayFirewallPolicyManagedRuleOverride : NetworkB
         public override void ExecuteCmdlet()
         {
             base.ExecuteCmdlet();
+
+            if (!this.MyInvocation.BoundParameters.ContainsKey("RuleId"))
+            {
+                this.RuleId = "";
```

Comment:
Is this really needed? What will the default value be?

Reply:
this is making the default value an empty string so the whole rule is disabled.

Do you think I should just make it empty in 'Rule' directly?

Reply:
Since RuleId field is mandatory, if we get here then RuleId will always be present right?
Since you made the RuleId array in the RuleGroup as optional, basically the list there will be empty. Then is this code really needed?

Reply:
Deleted this part.